core/src/net/network_thread.rs

```diff
@@ -1287,4 +1286,122 @@ mod tests {
         );
         network_thread.stop();
     }
+
+    // Creates 5 different network_threads, connects "1" to 2, 3, and 4 properly, then the 5th,
+    // asserts that the 2nd is disconnected, sends something on 3rd
+    // then the 2nd connects to 1, and we asserts that 4 is dropped by 1.
+    #[test]
+    fn channel_limit() -> () {
```

Comment:
I think we should have some kind of stress test in the integration tests, that just keeps opening new connections at a faster rate than they can be closed (if we can manage to do so without introducing new flaky tests.)